search/quality_eval_test.go

```diff
@@ -0,0 +1,315 @@
+			if _, err := testDB.Exec(statement); err != nil {
+				return fmt.Errorf("Unable to apply migration %s: %s\nStatement: %s\n", m.Name, err, statement)
+			}
+		}
+
+		return nil
+	}
+
+	return filepath.Walk("../migrations", visit)
+}
+
+func (suite *QualityEvalSuite) SetupSuite() {
+	err := suite.InitTestDB()
+	if err != nil {
+		panic(err)
+	}
+	suite.ctx = context.Background()
+
+	// Set package db.
+	common.InitWithDefault(suite.DB)
+	boil.DebugMode = viper.GetString("boiler-mode") == "debug"
+}
+
+func (suite *QualityEvalSuite) TearDownSuite() {
+	// Close connections.
+	common.Shutdown()
+	// Drop test database.
+	suite.Require().Nil(suite.DestroyTestDB())
+}
+
+// In order for 'go test' to run this suite, we need to create
+// a normal test function and pass our suite to suite.Run
+func TestEngine(t *testing.T) {
+	suite.Run(t, new(QualityEvalSuite))
+}
+
+func (suite *QualityEvalSuite) TestGetLatestUidByFilters() {
+	fmt.Printf("\n------ TestGetLatestUidByFilters ------\n\n")
+	r := require.New(suite.T())
+
+	var latestUID string
+	filmDateMask := "{\"film_date\":\"2018-01-%s\"}"
+
+	childTag := mdbmodels.Tag{Pattern: null.String{"ibur", true}, ID: 1, UID: "L2jMWyce"}
+	childSource := mdbmodels.Source{Pattern: null.String{"bs-akdama-zohar", true}, ID: 1, TypeID: 1, UID: "ALlyoveA"}
+	parentTag := mdbmodels.Tag{Pattern: null.String{"arvut", true}, ID: 2, UID: "L3jMWyce"}
+	parentSource := mdbmodels.Source{Pattern: null.String{"bs-akdmot", true}, ID: 2, TypeID: 1, UID: "1vCj4qN9"}
+
+	r.Nil(suite.updateTagParent(childTag, parentTag, true, true))
+	r.Nil(suite.updateSourceParent(childSource, parentSource, true, true))
+
+	for i := 1; i < 13; i++ {
+		properties := null.JSON{JSON: []byte(fmt.Sprintf(filmDateMask, strconv.Itoa(i))), Valid: true}
+		cuUID, err := suite.addContentUnitTag(mdbmodels.ContentUnit{Properties: properties, Secure: 0, Published: true, ID: int64(i)}, consts.LANG_ENGLISH, childTag)
+		r.Nil(err)
+		_, err = suite.addContentUnitSource(mdbmodels.ContentUnit{UID: cuUID}, consts.LANG_ENGLISH, childSource)
+		r.Nil(err)
+		latestUID = cuUID
+	}
+
+	sourceFilter := search.Filter{Name: search.FILTER_NAME_SOURCE, Value: parentSource.UID}
+	tagsFilter := search.Filter{Name: search.FILTER_NAME_TOPIC, Value: parentTag.UID}
+
+	fmt.Printf("Test by source \n")
+	resultUID, err := search.GetLatestUidByFilters([]search.Filter{sourceFilter}, common.DB)
+	r.Nil(err)
+	r.Equal(latestUID, resultUID)
+
+	fmt.Printf("Test by topic \n")
+	resultUID, err = search.GetLatestUidByFilters([]search.Filter{tagsFilter}, common.DB)
+	r.Nil(err)
+	r.Equal(latestUID, resultUID)
+
+	fmt.Printf("Test by source and topic \n")
+	resultUID, err = search.GetLatestUidByFilters([]search.Filter{sourceFilter, tagsFilter}, common.DB)
+	r.Nil(err)
+	r.Equal(latestUID, resultUID)
+}
+
+func (suite *QualityEvalSuite) updateSourceParent(child mdbmodels.Source, parentSource mdbmodels.Source, insertChild bool, insertParent bool) error {
```

Comment:
Don't we have those functions already here:
https://github.com/Bnei-Baruch/archive-backend/blob/master/es/indexer_test.go#L273

What is new that we use only here?
Maybe we extract those to common lib?
WDYT?

Reply:
I mean also the two functions below.